tests/py2puml/inspection/test_inspectclass.py

```diff
@@ -148,3 +149,49 @@ def test_inspect_module_should_unwrap_decorated_constructor():
     # PointDecoratedWithoutWrapping UmlClass
     point_without_wrapping_umlitem: UmlClass = domain_items_by_fqn['tests.modules.withwrappedconstructor.PointDecoratedWithoutWrapping']
     assert len(point_without_wrapping_umlitem.attributes) == 0, 'the attributes of the original constructor could not be found, the constructor was not wrapped by the decorator'
+
+
+def test_inspect_module_should_find_methods():
+    """ Test that methods are detected including static methods """
+
+    domain_items_by_fqn: Dict[str, UmlItem] = {}
+    domain_relations: List[UmlRelation] = []
+    inspect_module(
+        import_module('tests.modules.withmethods.withmethods'),
+        'tests.modules.withmethods.withmethods',
+        domain_items_by_fqn, domain_relations
+    )
+
+    # Coordinates UmlClass
+    coordinates_umlitem: UmlClass = domain_items_by_fqn['tests.modules.withmethods.withmethods.Coordinates']
+    assert len(coordinates_umlitem.methods) == 0
+
+    # Point UmlClass
+    point_umlitem: UmlClass = domain_items_by_fqn['tests.modules.withmethods.withmethods.Point']
+    assert len(point_umlitem.methods) == 3
+
+    assert point_umlitem.methods[0].name == 'from_values'
+    assert point_umlitem.methods[1].name == 'get_coordinates'
+    assert point_umlitem.methods[2].name == '__init__'
```

Comment:
we should also assert the return types of the methods

Reply:
Yes, but the return type is not available yet in the dataclass UmlMethod. I suggest to update the test case once this class has been fleshed out.

Reply:
agreed :+1: